.github/workflows/build.yml

```diff
@@ -1,15 +1,20 @@
-name: Build
+name: Build and tag nightlies
 
-on: [push]
+on:
+  push:
+    branches:
+      - master
 
 jobs:
   build:
-    runs-on: ubuntu-18.04
+    name: Compile & Generate Tagged Nightly
+    runs-on: ubuntu-22.04
     env:
       OUTPUT_PATH: output/get5
       OUTPUT_SM_PATH: output/get5/addons/sourcemod
       SM_JSON_INC_PATH: get5/dependencies/sm-json/addons/sourcemod/scripting/include
       STEAMWORKS_URL: https://raw.githubusercontent.com/KyleSanderson/SteamWorks/1.2.3c/Pawn/includes/SteamWorks.inc
+      GET5_VERSION: 0.8.0
```

Comment:
Shouldn't this be read from commit tag?

Reply:
I kind of threw it in here because this would also avoid a headache of trying to regex out the version number. Maybe we should put it in here and place that in the `version.sp` file instead?

Reply:
So you want to have this document be the source of truth for the version, not the tags?

Reply:
I was thinking possibly use this as a single point for the version number, yes.

Right now the only place that the true version exists is in the `version.sp` which is a 4 line file and that was updated to point to new version releases. This is essentially just moving it to another file for further automation, where nightly tags (which would build, and attach the `.zip` and `.tar.gz` files to the tag) can be generated on each push to the dev branch.

I feel this has benefits because then if a build fails when it's pushed to dev, a tag will not be created for that nightly or built until it is resolved. For tagging nightlies, I was thinking we could have the tags as something like `v0.8.0-{short_commit_hash}`, and for full stable releases, we can automate a draft release that will just be `v0.9.0` (or whichever version number we're at when we are stable), when pushed to master. I still need to workout the logic for this, but I was thinking this may be a better solution for CI purposes while being able to keep a bleeding edge nightly that can exist for greater than 90 days like the current release, and have a way of drafting stable releases quickly. I think we can even scrap the "Pre-Release" section that GitHub has as well, as I believe tags can also have attachments, too.